modules/qna/src/views/full/FormModal.jsx

```diff
@@ -151,20 +159,35 @@ export default class FormModal extends Component {
     )
   }
 
-  handleSubmit = event => {
+  handleSubmit = async event => {
     event.preventDefault()
     if (this.validateForm()) {
       return
     }
 
+    await this.handleDeletedContentElements()
     const item = this.state.item
     const questions = this.trimItemQuestions(item.questions)
 
     this.props.modalType === 'edit' ? this.onEdit(questions) : this.onCreate(questions)
   }
 
-  createAnswer = answer => {
+  createAnswer = (answer, notNew) => {
     const answers = [...this.state.item.answers, answer]
+    //Handle contentElements
+    if (answer.contentId) {
+      //If this is a new contentElement it stores the id in the list to further usage
+      if (!notNew) {
+        this.setState({ createdContentElements: [...this.state.createdContentElements, answer.contentId] })
+      }
+      //will remove the contentElement from the deleted list if added again
+      const deletedContentElements = this.state.deletedContentElements
+      const deletedIndex = deletedContentElements.findIndex(c => c == answer.contentId)
+      if (deletedIndex != -1) {
+        deletedContentElements.splice(deletedIndex, 1)
+        this.setState({ deletedContentElements })
+      }
+    }
```

Comment:
I'd remove most part of this logic. I think QNA and Content picker should be separate. On the picker, you can either create element or edit them. If you pick them in the qna afterwards, then you "link" them